src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -333,41 +334,49 @@ object ApiOperations {
                               tables: FilteredTables,
                               filter: Filter): Option[JoinedTables] = {
 
-    val blocks = tables.filteredBlocks
-    val operationGroups = tables.filteredOperationGroups
-    val operations = tables.filteredOperations
-    val accounts = tables.filteredAccounts
+    val FilteredTables(accounts, blocks, operationGroups, operations)= tables
+
+    def unwrapQuadJoin[A,B,C,D](nest: (((A,B),C),D)): (A,B,C,D) = nest match {
```

Comment:
So slick doesn't care about nested parentheses in type signatures when executing them? That would be such a relief.

Reply:
Well I'm not sure how to read your comment.

Slick do care about the nesting when it handles data, because it determines the _shape_ of input and output values.

But _we_ don't care in the end, because the results of queries can always be mapped to a more useful format, just like when you use it to make projections on the columns using `select a, b, c from ...`

When joining the resulting rows are packed within nested tuples which are probably more amenable to be composed over if you need to bolt on other joins later, this is my guess at why the library authors did it like this.

But for our goals, it's easier to put all row types in a single flattened tuple. We do the mapping only at the end, because I don't think the `join` signatures would match so easily anymore.